test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/SubscriptionSetTests.kt

```diff
@@ -0,0 +1,321 @@
+            user,
+            TestHelper.randomPartitionValue(),
+            setOf(FlexParentObject::class, FlexChildObject::class)
+        )
+        Realm.open(config).use { partionBasedRealm ->
+            assertFailsWith<IllegalStateException> { partionBasedRealm.subscriptions }
+        }
+    }
+
+    @Test
+    fun subscriptions_throwsOnClosedRealm() {
+        realm.close()
+        assertFailsWith<IllegalStateException> { realm.subscriptions }
+    }
+
+    @Test
+    fun initialSubscriptions() {
+        val subscriptions = realm.subscriptions
+        assertEquals(0, subscriptions.size)
+        assertEquals(SubscriptionSetState.PENDING, subscriptions.state)
+    }
+
+    @Test
+    fun findByQuery() = runBlocking {
+        val query = realm.query<FlexParentObject>()
+        val subscriptions = realm.subscriptions
+        assertNull(subscriptions.findByQuery(query))
+        subscriptions.update { add(query) }
+        val sub: Subscription = subscriptions.findByQuery(query)!!
+        assertNotNull(sub)
+        assertEquals("FlexParentObject", sub.objectType)
+    }
+
+    @Test
+    fun findByName() = runBlocking {
+        val subscriptions = realm.subscriptions
+        assertNull(subscriptions.findByName("foo"))
+        subscriptions.update {
+            realm.query<FlexParentObject>().subscribe("foo")
+        }
+        val sub: Subscription = subscriptions.findByName("foo")!!
+        assertNotNull(sub)
+        assertEquals("foo", sub.name)
+    }
+
+    @Test
+    fun state() = runBlocking {
+        val subscriptions = realm.subscriptions
+        assertEquals(SubscriptionSetState.PENDING, subscriptions.state)
+        subscriptions.update {
+            realm.query<FlexParentObject>().subscribe("test1")
+        }
+        assertEquals(SubscriptionSetState.PENDING, subscriptions.state)
+        subscriptions.waitForSynchronization()
+        assertEquals(SubscriptionSetState.COMPLETE, subscriptions.state)
+        subscriptions.update {
+            // `age` is not a queriable field
+            realm.query<FlexParentObject>("age > 42").subscribe("test2")
+        }
+        assertFailsWith<FlexibleSyncQueryException> {
+            subscriptions.waitForSynchronization()
+        }
+        assertEquals(SubscriptionSetState.ERROR, subscriptions.state)
+    }
+
+    @Test
+    fun size() = runBlocking {
+        val subscriptions = realm.subscriptions
+        assertEquals(0, subscriptions.size)
+        subscriptions.update {
+            realm.query<FlexParentObject>().subscribe()
+        }
+        assertEquals(1, subscriptions.size)
+        subscriptions.update {
+            removeAll()
+        }
+        assertEquals(0, subscriptions.size)
+    }
+
+    @Test
+    fun errorMessage() = runBlocking {
+        val subscriptions = realm.subscriptions
+        assertNull(subscriptions.errorMessage)
+        subscriptions.update {
+            realm.query<FlexParentObject>("age > 42").subscribe()
```

Comment:
Is this a bad syntax? or unsupported now from the server-side?

Reply:
`age` is not on the list of queriable fields configured on the server, so it will reject queries like this.